### PR TITLE
Patch to #867 and #881 about `mass_map`

### PR DESCRIPTION
**Pull request by panxiang126**
For previous discussions, see [#867](https://github.com/deepmodeling/dpgen/pull/867) and [#881](https://github.com/deepmodeling/dpgen/pull/881).

It is very strange that the comparison is with the version before [#881](https://github.com/deepmodeling/dpgen/pull/881).

### CODE REVIEW

File: dpgen/generator/run.py [hunk 1]
```diff
@@ -3662,6 +3661,8 @@ def run_iter (param_file, machine_file) :
        with open (machine_file, 'r') as fp:
            mdata = json.load (fp)
 
+    if jdata['mass_map'] == 'auto': jdata['mass_map'] = [get_atomic_masses(i) for i in jdata['type_map']]
```
Comment (panxiang126) [inline dpgen/generator/run.py:3664]: Please check if it is appropriate to modify jdata here.

**Comment by njzjz** [comment]
Please revert https://github.com/deepmodeling/dpgen/pull/881 in your PR.

**Comment by panxiang126** [comment]
> Please revert #881 in your PR.

I see. it is done.

**Comment by njzjz** [comment]
Do you still want to update [dpgen/tools/relabel.py](https://github.com/deepmodeling/dpgen/pull/881/files#diff-b2ded82a6e56dcad7640f4441a22a6703cb5e0ce829cc9962ed1209a6358914f)? I see changes to this file are in #867 but not in this PR.

File: dpgen/generator/run.py (hunk 1 above)
Comment (njzjz) [reply dpgen/generator/run.py:3664]: Please do not merge two lines into one line.

Comment (njzjz) [reply dpgen/generator/run.py:3664]: I think what @wanghan-iapcm suggests is to make a new method named `update_mass_map`.

Comment (njzjz) [reply dpgen/generator/run.py:3664]: Also, a unit test can be added for `update_mass_map`.

File: dpgen/generator/run.py [hunk 2]
```diff
@@ -3734,6 +3736,109 @@ def run_iter (param_file, machine_file) :
...
+        return atomic_masses[element_names.index(atom)]
+    elif atom in chemical_symbols:
+        return atomic_masses[chemical_symbols.index(atom)]
+    elif atom in atomic_number:
+        return atomic_masses[atomic_number.index(atom)]
+    else:
+        raise RuntimeError('unknown atomic identifier', atom, 'if one want to use isotopes, or non-standard element names, chemical symbols, or atomic number in the type_map list, please customize the mass_map list instead of using "auto".')
+
+
+def update_mass_map(jdata):
```
Comment (panxiang126) [inline dpgen/generator/run.py:3837]: add `update_mass_map` here

File: dpgen/generator/run.py (hunk 1 above)
Comment (panxiang126) [reply dpgen/generator/run.py]: How to add a unit test?

**Comment by panxiang126** [comment]
> Do you still want to update [dpgen/tools/relabel.py](https://github.com/deepmodeling/dpgen/pull/881/files#diff-b2ded82a6e56dcad7640f4441a22a6703cb5e0ce829cc9962ed1209a6358914f)? I see changes to this file are in #867 but not in this PR.

updated.

**Review by wanghan-iapcm — APPROVED** [review]

File: dpgen/generator/run.py (hunk 1 above)
Comment (wanghan-iapcm) [reply dpgen/generator/run.py]: you may check https://github.com/deepmodeling/dpgen/tree/master/tests/generator/test*py for examples

**Review by njzjz — APPROVED** [review]

**Comment by njzjz** [comment]
Please resolve conflicts.

**Pull request merged**